### ISSUE TITLE
`scss/declaration-nested-properties` wrongly reports some selectors

### ISSUE THREAD

**Issue by ArmorDarks** [kristerkari/stylelint-scss#143]
Seems that following rule doesn't work well with escaping in selectors,

```yml
  scss/declaration-nested-properties: never
```

This

```scss
.test4\:3 {}
```

Will produce following errors:

```shell
source/styles/generic/_code.scss
 1:1  ×  Unexpected nested property ".test4\"                             scss/declaration-nested-properties
```

![sublime_text_2017-07-18_19-27-44](https://user-images.githubusercontent.com/4460311/28328418-34728d14-6bef-11e7-8a25-2cf2a0e0f852.png)

**Comment by kristerkari** [comment]
Thanks for reporting! This rule also needs to handle escaped characters.

**Comment by ArmorDarks** [comment]
Thanks, works perfectly!
